--- 05_logging.cfg	2018-03-27 18:49:38.000000000 +0000
+++ 05_logging.cfg.new	2019-01-30 22:36:12.368710977 +0000
@@ -38,32 +38,19 @@
    format=%(asctime)s - %(filename)s[%(levelname)s]: %(message)s
    
    [formatter_simpleFormatter]
-   format=[CLOUDINIT] %(filename)s[%(levelname)s]: %(message)s
- - &log_file |
-   [handler_cloudLogHandler]
-   class=FileHandler
-   level=DEBUG
-   formatter=arg0Formatter
-   args=('/var/log/cloud-init.log',)
+   format=%(levelname)s %(filename)s: %(message)s
  - &log_syslog |
    [handler_cloudLogHandler]
    class=handlers.SysLogHandler
-   level=DEBUG
+   level=INFO
    formatter=simpleFormatter
-   args=("/dev/log", handlers.SysLogHandler.LOG_USER)
+   args=("/dev/log", handlers.SysLogHandler.LOG_DAEMON)
 
 log_cfgs:
 # Array entries in this list will be joined into a string
 # that defines the configuration.
 #
-# If you want logs to go to syslog, uncomment the following line.
-# - [ *log_base, *log_syslog ]
-#
-# The default behavior is to just log to a file.
-# This mechanism that does not depend on a system service to operate.
- - [ *log_base, *log_file ]
-# A file path can also be used.
-# - /etc/log.conf
+ - [ *log_base, *log_syslog ]
 
 # This tells cloud-init to redirect its stdout and stderr to
 # 'tee -a /var/log/cloud-init-output.log' so the user can see output
