diff --git a/systemd/cloud-config.service b/systemd/cloud-config.service
index 41a8614..d110b11 100644
--- a/systemd/cloud-config.service
+++ b/systemd/cloud-config.service
@@ -1,8 +1,8 @@
 [Unit]
 Description=Apply the settings specified in cloud-config
-After=network.target syslog.target cloud-config.target
+After=network-online.target syslog.target cloud-config.target
 Requires=cloud-config.target
-Wants=network.target
+Wants=network-online.target
 
 [Service]
 Type=oneshot
diff --git a/systemd/cloud-final.service b/systemd/cloud-final.service
index ef0f52b..7613d64 100644
--- a/systemd/cloud-final.service
+++ b/systemd/cloud-final.service
@@ -1,8 +1,8 @@
 [Unit]
 Description=Execute cloud user/final scripts
-After=network.target syslog.target cloud-config.service rc-local.service
+After=network-online.target syslog.target cloud-config.service rc-local.service
 Requires=cloud-config.target
-Wants=network.target
+Wants=network-online.target
 
 [Service]
 Type=oneshot
diff --git a/systemd/cloud-init.service b/systemd/cloud-init.service
index 6b0c722..fbfd410 100644
--- a/systemd/cloud-init.service
+++ b/systemd/cloud-init.service
@@ -1,8 +1,8 @@
 [Unit]
 Description=Initial cloud-init job (metadata service crawler)
-After=local-fs.target network.target cloud-init-local.service
+After=local-fs.target network-online.target cloud-init-local.service
 Before=sshd.service sshd-keygen.service
-Requires=network.target
+Requires=network-online.target
 Wants=local-fs.target cloud-init-local.service sshd.service sshd-keygen.service
 
 [Service]
