--- configure	2020-05-13 03:32:50.000000000 +0000
+++ configure.werror	2020-09-01 17:16:23.547647381 +0000
@@ -7225,14 +7225,14 @@
 
 if test "$ICC" = "yes"
 then
-      CFLAGS="$CFLAGS -diag-disable 187 -Wall -Werror"
+      CFLAGS="$CFLAGS -diag-disable 187 -Wall"
 
 $as_echo "#define _GNU_SOURCE 1" >>confdefs.h
 
 elif test "$GCC" = "yes"
 then
   GCC_VERSION=`$CC -dumpversion`
-  CFLAGS="$CFLAGS -Wall -Werror -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
+  CFLAGS="$CFLAGS -Wall -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
   if test "x$enable_asan" = "xyes"; then
     CFLAGS="$CFLAGS -fsanitize=address"
   fi
